### PR TITLE
WASM: NYI M128 globals + case type strenghtening

### PR DESCRIPTION
**Pull request by Cellule**
Make it explicit that we don't support Wasm m128 globals at this time.

Strengthen switch-cases with WasmTypes to make sure we always handled the main types.
This will cause compile errors when we add new types to make sure we correctly handled them everywhere

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/5245)

### CODE REVIEW

**Comment by Cellule** [comment]
Related to #5078 
FYI @arunetm

**Comment by Cellule** [comment]
Ping

File: lib/Runtime/Library/WebAssemblyEnvironment.cpp [hunk 1]
```diff
@@ -150,10 +150,7 @@ void WebAssemblyEnvironment::SetGlobalInternal(uint32 offset, T val)
 
 Wasm::WasmConstLitNode WebAssemblyEnvironment::GetGlobalValue(Wasm::WasmGlobal* global) const
 {
-    if (!global)
-    {
-        Js::Throw::InternalError();
-    }
+    AssertOrFailFast(global);
```
Comment (MikeHolman) [inline lib/Runtime/Library/WebAssemblyEnvironment.cpp]: nit: seems like a failfast is going too conservative, if global is null we'll just end up with a nullptr dereference

Comment (Cellule) [reply lib/Runtime/Library/WebAssemblyEnvironment.cpp]: The check is not very expensive (we've had it for so long) and I think I rather be conservative and also it documents the method that we don't allow nullptr for global

**Pull request merged**